lib/site-inspector.rb

```diff
@@ -127,37 +134,213 @@ def to_json
     to_hash.to_json
   end
 
-  def to_hash
+  def http
+    details = {
+      # site-inspector's best guess
+      canonical: uri.to_s,
+      canonical_protocol: https? ? :https : :http,
+      canonical_endpoint: www? ? :www : :root,
+
+      endpoints: endpoints
+    }
+
+    combos = details[:endpoints]
+
+    details[:live] = (
+      combos[:https][:www][:live] or
+      combos[:https][:root][:live] or
+      combos[:http][:www][:live] or
+      combos[:http][:root][:live]
+    )
+
+    details[:broken_root] = (
+      (combos[:https][:root][:status] == 0) and
+      (combos[:http][:root][:status] == 0)
+    )
+
+    details[:broken_www] = (
+      (combos[:https][:www][:status] == 0) and
+      (combos[:http][:www][:status] == 0)
+    )
+
+    details[:enforce_https] = (
+      (
+        (combos[:http][:www][:status] == 0) ||
+        (combos[:http][:www][:redirect_to_https])
+      ) and
+      (
+        (combos[:http][:root][:status] == 0) or
+        (combos[:http][:root][:redirect_to_https])
+      ) and
+      (
+        combos[:https][:www][:live] or
+        combos[:https][:root][:live]
+      )
+    )
+
+    details[:redirect] = (
+      combos[:http][:www][:redirect_to_external] and
+      combos[:http][:root][:redirect_to_external] and
+      combos[:https][:www][:redirect_to_external] and
+      combos[:https][:root][:redirect_to_external]
+    )
+
+    # HSTS on the canonical domain? (valid HTTPS checked in endpoint)
+    details[:hsts] = combos[:https][details[:canonical_endpoint]][:hsts]
+
+    # HSTS on the entire domain?
+    details[:hsts_entire_domain] = (
+      combos[:https][:root][:hsts] and
+      combos[:https][:root][:hsts_header].downcase.include?("includesubdomains")
+    )
```

Comment:
When measuring whether HSTS applies to an "entire domain", e.g. all of `whitehouse.gov`, the only thing that matters is whether `https://whitehouse.gov` has an enforced HSTS header that has the `includeSubDomains` directive. [HSTS directives are case-insensitive.](http://tools.ietf.org/html/rfc6797#section-6.1) 

Whether the HTTPS is valid (required for HSTS to take effect) is evaluated per-endpoint, in the `http_endpoint` method.